src/Sylius/Bundle/SettingsBundle/Manager/SettingsManager.php

```diff
@@ -71,29 +79,22 @@ class SettingsManager implements SettingsManagerInterface
      */
     protected $eventDispatcher;
 
-    /**
-     * @param SchemaRegistryInterface  $schemaRegistry
-     * @param ObjectManager            $parameterManager
-     * @param RepositoryInterface      $parameterRepository
-     * @param FactoryInterface         $parameterFactory
-     * @param Cache                    $cache
-     * @param ValidatorInterface       $validator
-     * @param EventDispatcherInterface $eventDispatcher
-     */
     public function __construct(
-        SchemaRegistryInterface $schemaRegistry,
-        ObjectManager $parameterManager,
-        RepositoryInterface $parameterRepository,
+        ServiceRegistryInterface $schemaRegistry,
+        ServiceRegistryInterface $resolverRegistry,
+        ObjectManager $settingsManager,
+        FactoryInterface $settingsFactory,
         FactoryInterface $parameterFactory,
-        Cache $cache,
+        SettingsResolverInterface $defaultResolver,
```

Comment:
We can leave the default resolver concept as is, but for me the best solution will be to move resolver resolving responsibility from settings manager to resolver registry. 

Eg. if I call `$resolverRegistry->get('foo')` and it does not have `foo` resolver, it should return the default one (it can be done as a decorator, so there's no need to write a custom registry). This way settings manager is not aware of that logic (and there's less dependencies! :tada:) and there's a single point of truth for getting a settings resolver.

Moreover, if we will go with that approach, we won't need to register `default` resolver in the registry, so the issue with unregistering and registering again will never happen :)